fe/fe-core/src/main/java/org/apache/doris/datasource/CatalogMgr.java

```diff
@@ -241,6 +251,8 @@ public void dropCatalog(DropCatalogStmt stmt) throws UserException {
             CatalogLog log = CatalogFactory.constructorCatalogLog(catalog.getId(), stmt);
             replayDropCatalog(log);
             Env.getCurrentEnv().getEditLog().logCatalogLog(OperationType.OP_DROP_CATALOG, log);
+
+            lastDBOfCatalog.remove(stmt.getCatalogName());
```

Comment:
Need to modify `removeCatalog()` too.
Because `removeCatalog()` will be called when replaying edit log